tensorflow/tensorboard
# rust: upgrade to latest `rules_rust`, `cargo-raze`

Summary:
We upgrade to latest `rules_rust` mainly for bazelbuild/rules_rust#572.
This requires upgrading `cargo-raze` to the newly released v0.9.2, plus
a herd of interdependent changes:

  - update to `rules_rust` latest
  - rename `@io_bazel_rules_rust` to `@rules_rust`
  - update Prost `protoc` wiring (not sure exactly why the old one
    doesn’t work anymore, but the new one is nicer, anyway)
  - update doc server to remove arch name, which also makes it work when
    compiled with `-c opt`

Test Plan:
Build changes only, so that `bazel test` passes and a `--load_fast`
smoke screen looks good suffices for correctness.

To see that the `openssl-sys` build spam is gone:

```
rm -rf bazel-out/k8-opt/bin/external/raze__openssl_sys__0_9_60  # ignore EACCES
bazel build @raze__openssl_sys__0_9_60//:openssl_sys
```

To test the doc server:

```
bazel run //tensorboard/data/server:rustboard_core_doc_server
```

…and navigate to the listed port.

wchargin-branch: rules-rust-20210203
